.pipelines/e2e-tests-template.yaml

```diff
@@ -52,7 +52,7 @@ jobs:
             export REGISTRY="${REGISTRY:-$(REGISTRY_NAME).azurecr.io/k8s/csi/secrets-store}"
 
             make e2e-test
-          displayName: "Run e2e tests"
+          displayName: "Run e2e tests with Released Version"
```

Comment:
This template is used for nightly as well as pr jobs. We should add a conditional to run this step only when it's a nightly job. For instance: 

https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/5d1596aeb8d15aad8fb3c81826f978ecca2aa4c6/.pipelines/nightly.yaml#L24

we run cluster upgrade tests only with nightly job. That parameter is false for PR jobs.

Reply:
Not adding conditional run as this will help test released chart (tgz) in release branch/pipeline.